age/tui.go

```diff
@@ -67,3 +71,149 @@ func readPassphrase(prompt string) ([]byte, error) {
 	defer fmt.Fprintf(out, "\r\n")
 	return term.ReadPassword(int(in.Fd()))
 }
+
+func Printf(format string, v ...interface{}) {
+	log.Printf("age: "+format, v...)
+}
+
+func Warningf(format string, v ...interface{}) {
+	log.Printf("age: warning: "+format, v...)
+}
+
+// ClearLine clears the current line on the terminal, or opens a new line if
+// terminal escape codes don't work.
+func ClearLine(out io.Writer) {
+	const (
+		CUI = "\033["   // Control Sequence Introducer
+		CPL = CUI + "F" // Cursor Previous Line
+		EL  = CUI + "K" // Erase in Line
+	)
+
+	// First, open a new line, which is guaranteed to work everywhere. Then, try
+	// to erase the line above with escape codes.
+	//
+	// (We use CRLF instead of LF to work around an apparent bug in WSL2's
+	// handling of CONOUT$. Only when running a Windows binary from WSL2, the
+	// cursor would not go back to the start of the line with a simple LF.
+	// Honestly, it's impressive CONIN$ and CONOUT$ work at all inside WSL2.)
+	fmt.Fprintf(out, "\r\n"+CPL+EL)
+}
+
+// WithTerminal runs f with the terminal input and output files, if available.
+// WithTerminal does not open a non-terminal stdin, so the caller does not need
+// to check stdinInUse.
+func WithTerminal(f func(in, out *os.File) error) error {
```

Comment:
Can you replace the copy of this function's body in `readPassphrase()` with a call to `WithTerminal`?

Reply:
Done